File: inserccion_22_09_2025/Inserccion.java
public class Inserccion {
    public static void main(String[] args) {
        int[] arr = {5, 2, 9, 1, 5, 6};
        insertionSort(arr);
        for (int num : arr) {
            System.out.print(num + " ");
        }
    }
    public static void insertionSort(int[] arr) {
        for (int i = 1; i < arr.length; i++) {
            int tem = arr[i];
            int j = i - 1;
            while (j >= 0 && arr[j] > tem) {
                arr[j + 1] = arr[j];
                j--;
            }
            arr[j + 1] = tem;
        }
    }
}

File: inserccion_22_09_2025/inserccion.cpp
#include <iostream>

void inserccion(int arr[], int n) {
    int i, key, j;
    for (i = 1; i < n; i++) {
        key = arr[i];
        j = i - 1;

        // Mover los elementos de arr[0..i-1], que son mayores que key,
        // a una posición adelante de su posición actual
        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
}
int main() {
    int a[5] = {42, 23, 17, 13, 5};
    inserccion(a, 5);
    for (int i = 0; i < 5; i++) {
        std::cout << a[i] << " ";
    }
    std::cout << std::endl;
    return 0;
}